File: WallPanelApp/src/main/java/de/rhuber/homedash/WelcomeActivity.java
package de.rhuber.homedash;

import android.content.Intent;
import android.os.Build;
import android.os.Bundle;
import android.support.v7.app.AppCompatActivity;
import android.view.Menu;
import android.view.MenuItem;
import android.util.Log;

public class WelcomeActivity extends AppCompatActivity {

    private final String TAG = BrowserActivity.class.getName();
    private static boolean startup = true;

    @Override
    public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        setContentView(R.layout.activity_welcome);
        getFragmentManager().beginTransaction().add(android.R.id.content,
                new SettingsActivity.GeneralPreferenceFragment()).commit();

        if (startup) {
            Log.i(TAG, "Starting Browser on Startup");
            startBrowserActivity();
        }

        startup = false;
    }

    @Override
    public boolean onCreateOptionsMenu(Menu menu) {
        // Inflate the menu; this adds items to the action bar if it is present.
        getMenuInflater().inflate(R.menu.menu_welcome, menu);
        return true;
    }

    private void startBrowserActivity() {
        Log.d(TAG, "startBrowserActivity Called");
        Log.i(TAG, String.valueOf(WallPanelService.getInstance()));
        String browserType = (new Config(this.getApplicationContext())).getAndroidBrowserType();
        Class targetClass;
        switch (browserType) {
            case "Native":
                Log.i(TAG, "Explicitly using native browser");
                targetClass = BrowserActivityNative.class;
                break;
            case "Legacy":
                Log.i(TAG, "Explicitly using legacy browser");
                targetClass = BrowserActivityLegacy.class;
                break;
            case "Auto":
            default:
                Log.i(TAG, "Auto-selecting dashboard browser");
                targetClass = (Build.VERSION.SDK_INT >= Build.VERSION_CODES.LOLLIPOP) ?
                        BrowserActivityNative.class
                        :
                        BrowserActivityLegacy.class;
                break;
        }
        startActivity(new Intent(getApplicationContext(), targetClass));
    }

    @SuppressWarnings("UnusedParameters")
    public void onLaunchDashboard(MenuItem mi) {
        Log.d(TAG, "onLaunchDashboard Called");
        startBrowserActivity();
    }

}

File: WallPanelApp/src/main/java/com/thanksmister/iot/wallpanel/ui/activities/WelcomeActivity.kt
/*
 * Copyright (c) 2018 ThanksMister LLC
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software distributed
 * under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package com.thanksmister.iot.wallpanel.ui.activities

import android.content.Intent
import android.os.Build
import android.os.Bundle
import android.view.View
import com.thanksmister.iot.wallpanel.R

import com.thanksmister.iot.wallpanel.persistence.Configuration
import dagger.android.support.DaggerAppCompatActivity

import timber.log.Timber
import javax.inject.Inject

class WelcomeActivity : DaggerAppCompatActivity() {

    @Inject
    lateinit var configuration: Configuration

    public override fun onCreate(savedInstanceState: Bundle?) {

        super.onCreate(savedInstanceState)

        setContentView(R.layout.activity_welcome)

        if (supportActionBar != null) {
            supportActionBar!!.show()
            supportActionBar!!.title = getString(R.string.app_name)
        }

        findViewById<View>(R.id.welcomeSettingsButton).setOnClickListener {
            configuration.setFirstTime(false)
            startActivity(Intent(this@WelcomeActivity, SettingsActivity::class.java))
            finish()
        }

        if (!configuration.isFirstTime) {
            startBrowserActivity()
        }
    }

    private fun startBrowserActivity() {
        val browserType = configuration.androidBrowserType
        Timber.i("startBrowserActivity browserType $browserType")
        val targetClass: Class<*>
        when (browserType) {
            Configuration.PREF_BROWSER_NATIVE -> {
                Timber.i( "Explicitly using native browser")
                targetClass = BrowserActivityNative::class.java
            }
            Configuration.PREF_BROWSER_LEGACY  -> {
                Timber.i("Explicitly using legacy browser")
                targetClass = BrowserActivityLegacy::class.java
            }
            Configuration.PREF_BROWSER_AUTO -> {
                Timber.i("Auto-selecting dashboard browser")
                targetClass = if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.LOLLIPOP)
                    BrowserActivityNative::class.java
                else
                    BrowserActivityLegacy::class.java
            }
            else -> {
                Timber.i("Auto-selecting dashboard browser")
                targetClass = if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.LOLLIPOP)
                    BrowserActivityNative::class.java
                else
                    BrowserActivityLegacy::class.java
            }
        }
        val intent = Intent(this@WelcomeActivity, targetClass)
        intent.flags = Intent.FLAG_ACTIVITY_CLEAR_TOP
        startActivity(intent)
        finish()
    }
}